app/migrations/versions/1c4c356aa44c_.py

```diff
@@ -0,0 +1,76 @@
+"""Adds search_view with indexes
+
+Revision ID: 1c4c356aa44c
+Revises: fc2eeba8e8e8
+Create Date: 2020-02-07 15:38:07.298134
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '1c4c356aa44c'
+down_revision = 'fc2eeba8e8e8'
+branch_labels = None
+depends_on = None
+
+trigger_tuples = [
+    #user table
+    ('user', 'email'),
+    ('user', '_phone'),
+    ('user', 'first_name'),
+    ('user', 'last_name'),
+]
+
+def upgrade():
+    conn = op.get_bind()
+
+    conn.execute(sa.sql.text('''
+        CREATE MATERIALIZED VIEW search_view AS (
+            SELECT
+                u.id,
+                u.email,
+                u._phone,
+                u.first_name,
+                u.last_name,
+                u.default_transfer_account_id,
+                to_tsvector(u.email) AS tsv_email,
+                to_tsvector(u._phone) AS tsv_phone,
+                to_tsvector(u.first_name) AS tsv_first_name,
+                to_tsvector(u.last_name) AS tsv_last_name
+            FROM "user" u
+        );
+    '''))
+
+    op.create_index(op.f('ix_search_view_id'), 'search_view', ['id'], unique=True)
+    op.create_index(op.f('ix_tsv_email'), 'search_view', ['tsv_email'], postgresql_using='gin')
```

Comment:
With a DB up to date with latest master migration, I got the following error when running `python manage.py db upgrade`

```
INFO  [alembic.runtime.migration] Running upgrade fc2eeba8e8e8 -> 1c4c356aa44c, empty message
Traceback (most recent call last):
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1249, in _execute_context
    cursor, statement, parameters, context
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
psycopg2.ProgrammingError: column "ix_tsv_email" does not exist


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 44, in <module>
    manager.run()
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/flask_script/__init__.py", line 417, in run
    result = self.handle(argv[0], argv[1:])
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/flask_script/__init__.py", line 386, in handle
    res = handle(*args, **config)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/flask_migrate/__init__.py", line 259, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/command.py", line 279, in upgrade
    script.run_env()
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/script/base.py", line 475, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
    module = load_module_py(module_id, path)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/util/compat.py", line 174, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "migrations/env.py", line 91, in <module>
    run_migrations_online()
  File "migrations/env.py", line 83, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/runtime/environment.py", line 846, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/runtime/migration.py", line 365, in run_migrations
    step.migration_fn(**kw)
  File "/Users/tristancole/PycharmProjects/SempoBlockchain/app/migrations/versions/1c4c356aa44c_.py", line 53, in upgrade
    op.create_index(op.f('ix_tsv_email'), 'search_view', ['ix_tsv_email'], postgresql_using='gin')
  File "<string>", line 8, in create_index
  File "<string>", line 3, in create_index
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/operations/ops.py", line 975, in create_index
    return operations.invoke(op)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/operations/base.py", line 345, in invoke
    return fn(self, operation)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/operations/toimpl.py", line 88, in create_index
    operations.impl.create_index(idx)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/ddl/impl.py", line 276, in create_index
    self._exec(schema.CreateIndex(index))
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/alembic/ddl/impl.py", line 134, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1050, in _execute_ddl
    compiled,
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1253, in _execute_context
    e, statement, parameters, cursor, context
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1473, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1249, in _execute_context
    cursor, statement, parameters, context
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) column "ix_tsv_email" does not exist

[SQL: CREATE INDEX ix_tsv_email ON search_view USING gin (ix_tsv_email)]
(Background on this error at: http://sqlalche.me/e/f405)
```

Reply:
Fully nuked my local database, and all the migrations ran fine, I'm also able to downgrade and upgrade on/off this migration freely. I _did_ just upgrade this migration in the last batch of changes though, so perhaps if you try again it'll work now.

Reply:
Seems to work now